Migrate external reset password test

Cruikshanks opened the pull request:
This migration highlighted that the previous tests didn't really understand the use of `it()` blocks and that they should be tests in isolation. Instead, they seemed to be more of a way of breaking up what was going on in the file.

In Cypress v12 `testIsolation` is enabled by default. This means Cypress [resets the browser context _before_ each test](https://docs.cypress.io/guides/references/migration-guide#Test-Isolation). Each `it()` block is considered a test. So, any that could not be run in isolation will now fail and this definitely applied to 'reset password'.

On top of that, it was again sending [axios requests](https://www.npmjs.com/package/axios) instead of using Cypress, and doing so to get an ID we hard code when setting up the test data. So we've been able to make some improvements, removing unnecessary dependencies (done in the initial commit) and streamlining the tests themselves.

Having looked at the existing tests, they were both checking validation and that the journey worked. So, we've used that as a basis for creating 2 tests that do the same thing but which can now be run in isolation.

timstone commented on features/check_licence_details.feature:
```diff
@@ -1,12 +1,12 @@
-@frontoffice
+@frontoffice @wip
```
Please remove @ wip tag

andrewhick commented:
Removed @wip as suggested

Cruikshanks committed:
Migrate external reset password test

This migration highlighted that the previous tests didn't really understand the use of `it()` blocks and that they should be tests in isolation. Instead they seemed to be more a way of breaking up what was going on in the file.

In Cypress v12 `testIsolation` is enabled by default. This means Cypress [resets the browser context _before_ each test](https://docs.cypress.io/guides/references/migration-guide#Test-Isolation). Each `it()` block is considered a test. So, any that could not be run in isolation will now fail and this definitely applied to reset password.

On top of that it was again sending [axios requests](https://www.npmjs.com/package/axios) instead of using Cypress, and doing so to get an ID we hard code when setting up the test data. So we've been able to make some improvements, removing unnecessary dependencies (done in the initial commit) and streamlining the tests themselves.

Having looked at the existing tests, they were both checking validation and that the journey worked. So, we've used that as a basis for creating 2 tests that do the same thing but which can now be run in isolation.
